metaflow/plugins/datatools/s3/s3.py

```diff
@@ -1760,17 +1761,33 @@ def try_s3_op(last_ok_count, pending_retries, out_lines, inject_failures):
                 # due to a transient failure so we try again.
                 transient_retry_count += 1
                 total_ok_count += last_ok_count
-                print(
-                    "Transient S3 failure (attempt #%d) -- total success: %d, "
-                    "last attempt %d/%d -- remaining: %d"
-                    % (
-                        transient_retry_count,
-                        total_ok_count,
-                        last_ok_count,
-                        last_ok_count + last_retry_count,
-                        len(pending_retries),
+
+                # Extract transient error type from pending retry lines
+                error_info = ""
+                if pending_retries:
+                    try:
+                        # Parse the first line to get transient error type
+                        first_retry = json.loads(
+                            pending_retries[0].decode("utf-8").strip()
+                        )
+                        if "transient_error_type" in first_retry:
+                            error_info = " (%s)" % first_retry["transient_error_type"]
+                    except (json.JSONDecodeError, IndexError, KeyError):
+                        pass
+
+                if S3_LOG_TRANSIENT_RETRIES:
```

Comment:
Can we move the above code into this if block -- I don't think anything is used outside of this block.